includes/myparcel-shipment-hooks.php

```diff
@@ -37,15 +37,15 @@ function myparcelcomWebhookCallback(WP_REST_Request $request)
         }
     }
 
-    $order = wc_get_order((int) $shipmentData['attributes']['customer_reference']);
-    $shipmentId = get_post_meta($order->get_id(), MYPARCEL_SHIPMENT_ID, true);
+    $order = wc_get_order((int) $shipmentData['attributes']['customer_reference'] ?? null);
+    $shipmentId = get_post_meta((int) $order?->get_id(), MYPARCEL_SHIPMENT_ID, true);
 
     if ($shipmentId === $shipmentData['id']) {
         update_post_meta($order->get_id(), MYPARCEL_SHIPMENT_DATA, json_encode([
```

Comment:
no need to use the null check operator and cast to `int` here?
```suggestion
        update_post_meta((int) $order?->get_id(), MYPARCEL_SHIPMENT_DATA, json_encode([
```

Reply:
Oh wait, I misread your comment. No if line 41 does not find the order, it will not end up here :)